### ISSUE TITLE
Kernel panic when writing output to disk

### ISSUE THREAD

**Issue by canergen** [dcjones/proseg#12]
Hi, thanks for the amazing tool. It works well on very different datasets. However, I have a single dataset, where I get kernel panicking when writing output to disk (reproducible over 4 segmentation runs and after redownloading the data).
The error message is cryptic to understand, what is going wrong. I will now tile this dataset into smaller chunks but it would be great to get some feedback, on what is going wrong. It would also be nice to be able to checkpoint runs, so when it crashes it doesn't start from scratch (most crashes are memory related but this one would be weird - it uses 200 GB memory RAM most of the time but the server has 900 GB of RAM).

```
Using 40 threads
Read 476710604 transcripts
     308796 cells
     1000 genes
Estimated full area: 5484591
Full volume: 383921380
Using grid size 43.659657. Chunks: 3182
00:00:00 ############################################################ | log-likelihood: -2326479600 | assigned: 474549634 / 476710604 (99.55%) | non-background: (96.21%)[/home/eecs/cergen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proseg-1.0.4/src/output.rs:385:5] fovs.len() = 476710604
[/home/eecs/cergen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proseg-1.0.4/src/output.rs:386:5] fov_names.len() = 383
[/home/eecs/cergen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proseg-1.0.4/src/output.rs:387:5] transcripts.len() = 476710604
thread 'main' panicked at /home/eecs/cergen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow2-0.18.0/src/array/physical_binary.rs:208:41:
called `Result::unwrap()` on an `Err` value: Overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by dcjones** [comment]
I'm glad proseg is mostly working well. Can you tell me what version you are using (by running `proseg --version`). I fixed a similar bug a while ago, but I'm not sure if it's that one or a related one.

**Comment by canergen** [comment]
`proseg --version: proseg 1.0.4`

**Comment by canergen** [comment]
Maybe the most essential information, it's when using CosMX data without converting the data beforehand.
`proseg cancer_tx_file.csv --cosmx `

**Comment by dcjones** [comment]
Ok, thanks. One other question:  can you see which output files were generated? There's some kind of overflow error while it's generating the output files, but I can't tell for sure which file from that error.

**Comment by canergen** [comment]
The folder contains (I have no idea whether they are complete):
`cancer_tx_file.csv  cell-metadata.csv.gz  expected-counts.csv.gz`

**Comment by dcjones** [comment]
Ok, I just published version 1.0.5 which I think fixes this issue. Give it a shot and let me know when you have a chance.

**Comment by canergen** [comment]
Thanks, the fix is working.
